pkg/gc/config.go

```diff
@@ -81,3 +82,16 @@ func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
 
 	return &c, nil
 }
+
+// SanityCheckedConfigMapLoader provides sanity checking for minRevisionTimeout and returns an error if this value is invalid
```

Comment:
Can we just change NewConfigFromConfigMap to add this validation?  This is generally where we do this sort of validation.

Reply:
The issue is ResyncPeriod is a variable and we need to supply a constructor to the ConfigStore which is of type `func(*k8s.io/api/core/v1.ConfigMap) (*Config , error)`. The alternative way I can see Is to make NewConfigFromConfigMap a method on a struct which stores the ResyncPeriod.

Reply:
I am curious why `NewConfigFromConfigMap` needs this signature if the return value of this function does?  Can't this just be curried?  e.g. `time.Duration -> *corev1.ConfigMap -> (*Config, error)`?

Reply:
Curry made. :)

Reply:
Can't wait for the haskell rewrite :)